### ISSUE TITLE
Unprocessable Entity when using GitLab Release

### ISSUE THREAD

**Issue by brunogirin** [release-it/release-it#533]
I'm trying to run `release-it` in GitLab CI, I followed [the docs](https://github.com/release-it/release-it/blob/master/docs/gitlab-releases.md) but when I run it in GitLab CI, I get:

```
✖ GitLab release
ERROR Response code 422 (Unprocessable Entity)
```

I am using release-it v12.2.2.

What can I do to debug that problem?

**Comment by webpro** [comment]
See https://github.com/release-it/release-it/blob/master/README.md#troubleshooting--debugging:

```
DEBUG=release-it:* release-it [...]
```

**Comment by asdptkt** [comment]
```
2019-06-25T08:56:52.830Z release-it:gitlab {
  url: 'https://alexandria.ggwp.red/api/v4/projects/sre%2Fmysqlinfo/releases',
  body: {
    name: 'Release 0.0.1',
    tag_name: '0.0.1',
    description: 'Release version v0.0.1-test.',
    assets: { links: [Array] }
  }
}
2019-06-25T08:56:52.908Z release-it:gitlab HTTPError: Response code 422 (Unprocessable Entity)
    at EventEmitter.<anonymous> (/usr/local/lib/node_modules/release-it/node_modules/got/source/as-promise.js:74:19)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
2019-06-25T08:56:52.916Z release-it:metrics {
  statusCode: 200,
  statusMessage: 'OK',
  url: 'http://www.google-analytics.com/collect'
}
ERROR Response code 422 (Unprocessable Entity)
ERROR: Job failed: exit code 1
```

Request body sent by release-it is missing "id" property in the JSON.

https://docs.gitlab.com/ee/api/releases/#create-a-release

**Comment by webpro** [comment]
Thanks @asdptkt. The `id` is the `owner%2Frepo` in the url.

I think the issue is in the `tag_name`. Somehow the `-test` is stripped off when bumping, likely here: https://github.com/release-it/release-it/blob/b2223f1d0a182e8e9ffe0280a6e1d35a773b4d32/lib/plugin/GitBase.js#L32

Will really dig into this as soon as I have the time.

**Comment by webpro** [comment]
Unfortunately, I can't reproduce the issue. Releases are created properly, regardless of the `tag_name`. Can someone please share more details, ideally a repo/configuration I can analyze?

@brunogirin @asdptkt

**Comment by asdptkt** [comment]
@webpro Here's what I got:

`release-it --ci --no-git --config release-it.yml`

Content of `release-it.yml`

```yaml
---

gitlab:
  release: true
  releaseNotes: echo "Release version $CI_COMMIT_TAG."
  assets:
    - mysqlinfo
    - config.example.yaml
    - Changelog.md
```

**Comment by webpro** [comment]
@asdptkt  With `--no-git`, the `git tag` command is skipped. No `increment` or version is provided, and `patch` is the default increment in `--ci` mode. This means that the version, and thus the (formatted) `tag_name` will be based on the latest tag (e.g. `v0.0.1-test` will be incremented to `0.0.1`).

So what you probably want is to add the version to the command:

```
release-it $CI_COMMIT_TAG --ci --no-git --config release-it.yml
```

You might also need to configure the `v` to be prepended to the parsed version:

```
git:
  tagName: v${version}
```

**Comment by asdptkt** [comment]
@webpro good to know that I probably just using this wrong, I'm gonna try it again and report back here. Thanks a lot.

**Comment by webpro** [comment]
I think what is missing in the GitLab story, is that release-it is fundamentally based on incrementing an existing tag/version. And your use case seemingly needs to skip that part. Coincidentally, with v12.3.1, you can also use what was just discussed in #537, which is better than my initial suggestion:

```
release-it --no-increment --ci --no-git --config release-it.yml
```

**Comment by webpro** [comment]
Closing due to inactivity

**Comment by krumka** [comment]
Hi, I've encountered the same error and found out that the issue was from the fact that I didn't set push: true in the release-it config : 
```
{
  "git": {
    "push": true
  }
}
```
The tag was non existent and gitlab could not create the release. I found the fix on the #645 but thought it would be nice to have it as a solution here.

And @webpro:

> @asdptkt With --no-git, the git tag command is skipped. No increment or version is provided, and patch is the default increment in --ci mode. This means that the version, and thus the (formatted) tag_name will be based on the latest tag (e.g. v0.0.1-test will be incremented to 0.0.1).

Are you saying that in any case with --ci we only got patch increment ?

And if so, in the doc, you are speaking about conventional commit bumping, could you point me the direction for achieving the bumping with conventional commit ?

Thanks !

**Comment by jscottSwift** [comment]
For anyone still running into this error with GitLab Releases:  

I ran into the same issue and discovered the same solution as krumka.  I was using the `--no-git.push` CLI option.  I thought that only referred to pushing commits.  Turns out it also applies to tags also.  So, in my use-case, I simply wanted to "cut a tag" and create a GitLab release based on the current version of the project.  So, my CLI options ended up being:
`--no-increment --no-git.commit --gitlab.release`
